pennylane/gradients/gradient_transform.py

```diff
@@ -292,11 +292,10 @@ def jacobian_wrapper(
             trainable_params = qml.math.get_trainable_indices(args)
 
             if interface == "jax" and argnum:
-                warnings.warn(
-                    "argnum is deprecated with the Jax interface. You should use argnums instead."
+                raise qml.QuantumFunctionError(
+                    "argnum is deprecated with the Jax interface. You should use argnums "
+                    "instead."

```

Comment:
At some point will we delete this logic?

Reply:
I think it not expensive to check and can save some users, I would remove it once autograd is gone